src/Mount.cpp

```diff
@@ -1603,6 +1768,126 @@ void Mount::enableAzAltMotors() {
 
 #endif
 
+
+#if (FOCUS_STEPPER_TYPE != STEPPER_TYPE_NONE)
+/////////////////////////////////
+//
+// isRunningFocus
+//
+/////////////////////////////////
+bool Mount::isRunningFocus() const {
+  return _stepperFocus->isRunning();
+}
+
+/////////////////////////////////
+//
+// getFocusSpeed
+//
+/////////////////////////////////
+float Mount::getFocusSpeed() const {
+  return _stepperFocus->speed();
+}
+
+/////////////////////////////////
+//
+// focusSetSpeedByRate
+//
+/////////////////////////////////
+void Mount::focusSetSpeedByRate(int rate)
+{
+    _focusRate = clamp(rate, 1, 4);
+    float speedFactor[] = { 0, 0.05, 0.2, 0.5, 1.0};
+    _maxFocusRateSpeed = speedFactor[_focusRate] * _maxFocusSpeed;
+    LOGV3(DEBUG_MOUNT,F("Mount::focusSetSpeedByRate: rate is %d -> %f"),_focusRate , _maxFocusRateSpeed );
+    _stepperFocus->setMaxSpeed(_maxFocusRateSpeed);
+
+    if (_stepperFocus->isRunning()) {
+      LOGV1(DEBUG_MOUNT,F("Mount::focusSetSpeedByRate: stepper is already running so adjust speed"));
+      //_stepperFocus->setSpeed(speedFactor[_focusRate ] * _maxFocusSpeed);
+    }
+}
+
+/////////////////////////////////
+//
+// focusContinuousMove
+//
+/////////////////////////////////
+void Mount::focusContinuousMove(int direction)
```

Comment:
We should definitely have a direction enum, currently you could call this with any number which would change the applied speed

Reply:
Done.